### PR TITLE
New UX for HTTPS Everywhere: For Release

### PR DESCRIPTION
**Pull request by zoracon**
Anyone reviewing: Please leave comments/feedback for code efficiency, potential mistakes, and semantics only. Design specs have been determined already.

### CODE REVIEW

File: chromium/pages/popup/style.css [hunk 1]
```diff
@@ -1,128 +1,316 @@
...
+  margin: 10px 0 0 0;
 }
 
-/* Underline rules that have notes. */
-.rule[title] span {
-  border-bottom: 1px dotted;
-  cursor: help;
+.down {
+  transform: rotate(45deg);
+  -webkit-transform: rotate(45deg);
```
Comment (Bisaloo) [inline chromium/pages/popup/style.css]: Is the `-webkit-` prefix really necessary? https://caniuse.com/#feat=transforms2d

File: chromium/pages/popup/style.css [hunk 2]
```diff
@@ -1,128 +1,316 @@
-html {
-  background-color: #fff
+:root {
+  --https-blue: #0a84ff;
+  --darker-blue: #0060df;
+  --text-main: #000;
+  --text-secondary: #464646;
+  --ligh-grey: #ececec;
```
Comment (Bisaloo) [inline chromium/pages/popup/style.css]: light-grey?

**Comment by Bisaloo** [comment]
I don't know if that counts as design specs. Feel free to ignore if you wish:

I find it slightly confusing that clicking on the `See more` text doesn't do anything and that you have to click on the arrow to actually see more info.

File: chromium/pages/popup/index.html [hunk 3]
```diff
@@ -11,78 +11,100 @@
...
-      <br>
-      <a id="donateEFF" href="https://supporters.eff.org/donate/support-https-everywhere" target="_blank" data-i18n="menu_donate_eff_imperative"></a>
-      <p>
-        <small>
-          <span data-i18n="about_version">Version</span>: <span id="current-version"></span>
-          <span id="rulesets-versions">
-          </span>
-        </small>
-      </p>
+      <a class="button" id="viewAllRules" href="https://www.eff.org/https-everywhere/atlas" target="_blank" data-i18n="menu_viewAllRules"></a>
```
Comment (Bisaloo) [inline chromium/pages/popup/index.html]: See related #17674

**Comment by pipboy96** [comment]
@zoracon The switch has a weird animation while toggling, see https://pipboy96.github.io/switcherror.webm.

**Comment by pipboy96** [comment]
Same issue (https://github.com/EFForg/https-everywhere/pull/17854#issuecomment-487902490) applies to FF.

**Comment by zoracon** [comment]
> I don't know if that counts as design specs. Feel free to ignore if you wish:
> 
> I find it slightly confusing that clicking on the `See more` text doesn't do anything and that you have to click on the arrow to actually see more info.
Agreed, I got feedback about that and forgot to incorporate it. Will work on that

File: chromium/pages/popup/style.css (hunk 2 above)
Comment (zoracon) [reply chromium/pages/popup/style.css]: Are you asking about why it's called that? Not sure what the question is here.

File: chromium/pages/popup/style.css (hunk 1 above)
Comment (zoracon) [reply chromium/pages/popup/style.css]: Probably not, old habits from previous web development days. I'll amend this.

**Comment by zoracon** [comment]
I see this failed quite a few testing cases due to the shifting around and some deletion, I will look through these again to clean up more stray and/or unneeded code. Thanks for reviewing so far.

**Comment by zoracon** [comment]
> @zoracon The switch has a weird animation while toggling, see https://pipboy96.github.io/switcherror.webm.

By "weird", do you mean that slide affect shouldn't have that after effect? I can try different options.

**Comment by pipboy96** [comment]
@zoracon Yes, I mean it shouldn't stretch like that.

**Comment by pipboy96** [comment]
Can we make refresh "forgiving", e.g. if user turns the switch on and back off (or vice versa) very quicky, there will be no refresh?

File: chromium/pages/popup/style.css (hunk 2 above)
Comment (pipboy96) [reply chromium/pages/popup/style.css]: @zoracon They probably think it's a typo, which it indeed looks like.

**Comment by zoracon** [comment]
> Can we make refresh "forgiving", e.g. if user turns the switch on and back off (or vice versa) very quicky, there will be no refresh?

If I get the time, I can look into that as well.

File: chromium/pages/popup/style.css (hunk 2 above)
Comment (zoracon) [reply chromium/pages/popup/style.css]: Oh, I see now. Thanks for catching that

File: chromium/pages/popup/style.css [hunk 4]
```diff
@@ -1,128 +1,307 @@
...
-  margin-left: 1em;
-  margin-right: 1em;
-  margin-top: 0;
-  margin-bottom: 0;
-  min-width: 23em;
+  color: var(--text-secondary);
   font-family: 'Lucida Grande', 'Segoe UI', Tahoma, 'DejaVu Sans', Arial, sans-serif;
-  font-size: 75%;
-  color: #303942;
+  margin: var(--margin);
```
Comment (pipboy96) [inline chromium/pages/popup/style.css]: I don't see `--margin` defined anywhere.

File: chromium/pages/popup/style.css [hunk 5]
```diff
@@ -1,128 +1,307 @@
...
+  opacity: 0;
+  overflow: hidden;
+  transition: 0s;
+}
+
+.show {
+  visibility: visible;
+  opacity: 1;
+  transition: visibility 0s linear 0s, opacity 300ms;
+}
```
Comment (pipboy96) [inline chromium/pages/popup/style.css]: Add `\n` here please.

**Comment by pipboy96** [comment]
See also #299.

**Comment by pipboy96** [comment]
@zoracon Please run `stylelint` and `eslint` over this when finished.

File: chromium/pages/popup/index.html [hunk 6]
```diff
@@ -11,78 +11,100 @@
...
-      <p>
-        <small>
-          <span data-i18n="about_version">Version</span>: <span id="current-version"></span>
-          <span id="rulesets-versions">
-          </span>
-        </small>
-      </p>
+      <a class="button" id="viewAllRules" href="https://atlas.eff.org/index.html" target="_blank" data-i18n="menu_viewAllRules"></a>
+      <a class="button" id="aboutTitle" href="https://www.eff.org/https-everywhere" target="_blank" data-i18n="about_title"></a>
+      <a class="button" id="donateEFF" href="https://supporters.eff.org/git donate/support-https-everywhere" target="_blank" data-i18n="menu_donate_eff_imperative"></a>
```
Comment (pipboy96) [inline chromium/pages/popup/index.html]: Typo here (`git donate`).

File: src/chrome/locale/en/https-everywhere.dtd [hunk 7]
```diff
@@ -8,7 +8,7 @@
 <!ENTITY https-everywhere.menu.donate_eff_imperative "Donate to EFF">
 <!ENTITY https-everywhere.menu.observatory "SSL Observatory Preferences">
 <!ENTITY https-everywhere.menu.globalEnable "HTTPS Everywhere is ON">
-<!ENTITY https-everywhere.menu.encryptAllSitesEligible "Encrypt All Sites Eligible (EASE)">
+<!ENTITY https-everywhere.menu.encryptAllSitesEligible "Encrypt All Sites Eligible is OFF">
```
Comment (pipboy96) [inline src/chrome/locale/en/https-everywhere.dtd]: You would also need the one for `Encrypt All Sites Eligible is ON`.

**Pull request merged**